### PR TITLE
[MINOR] New API additions and Controller Token addition

### PR DESCRIPTION
**Pull request by SaravanakumarR2018**
Review comments taken care

Plus the API addition has been taken care

Also the Cluster Addition through Token is taken care

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -178,6 +210,35 @@ func EndpointWatcher(api *KubernetesAPIServer, contr Controller) {
         go controller.Run(stop)
         return
 }
+
+func GetKubeEndpointsAll(api *KubernetesAPIServer) *v1.EndpointsList {
+	fmt.Println("SECRET GET ALL API: Calling kubernetes API server")
```
Comment (janraj) [inline pkg/controller/controller.go]: Correct the log messages to End Point get all

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -178,6 +210,35 @@ func EndpointWatcher(api *KubernetesAPIServer, contr Controller) {
...
+	obj, err := api.Client.Core().Endpoints(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		panic(err.Error())
+	}
+	return obj
+}
+
+
+func GetKubeEndpointsNamespace(api *KubernetesAPIServer, namespace string) *v1.EndpointsList {
+	fmt.Println("SECRET GET ALL API: Calling kubernetes API server")
```
Comment (janraj) [inline pkg/controller/controller.go]: Correct the log message

File: pkg/interface/interface.go [hunk 3]
```diff
@@ -131,26 +133,337 @@ type endpoints struct {
...
     	WatchEvents []string 
 }
-var k8sEndpointPath string = "/cic/nitro/v1/config/endpoints"
+var k8sClientPath string = "/cic/nitro/v1/config"
+var k8sEndpointPath string = k8sClientPath + "/endpoints"
+var k8sServicePath string = k8sClientPath + "/services"
+var k8sIngressPath string = k8sClientPath + "/ingresses"
+var k8sPodPath string = k8sClientPath + "/pods"
+var k8sSecretPath string = k8sClientPath + "/secrets"
 var baseURL string = "/cic/nitro/v1/config"
```
Comment (janraj) [inline pkg/interface/interface.go]: Noe base URL is required ?

**Review by janraj — APPROVED** [review]

File: pkg/controller/controller.go (hunk 2 above)
Comment (SaravanakumarR2018) [reply pkg/controller/controller.go]: Fixed

File: pkg/interface/interface.go (hunk 3 above)
Comment (SaravanakumarR2018) [reply pkg/interface/interface.go]: Fixed

File: pkg/controller/controller.go (hunk 1 above)
Comment (SaravanakumarR2018) [reply pkg/controller/controller.go]: Fixed

**Pull request merged**